flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackend.java

```diff
@@ -298,6 +303,11 @@ private EmbeddedRocksDBStateBackend(
         checkArgument(
                 overlapFractionThreshold >= 0 && this.overlapFractionThreshold <= 1,
                 "Overlap fraction threshold of restoring should be between 0 and 1");
+
+        this.rocksDBMemoryFactory =
```

Comment:
If the default is `RocksDBMemoryFactory.DEFAULT`, can it ever be null?

Reply:
I was concerned about deserialization. Because the field isn't `final`, initialization in the default constructor is optional, so it might break in the future.
But ATM it can't be null. I'll remove the check.